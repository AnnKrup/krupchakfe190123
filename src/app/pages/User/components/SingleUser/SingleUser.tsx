import * as React from 'react';
import { User } from '../../../HomePage/Features/Users/slice/types';
import { FC } from 'react';

type Props = {
  user: User;
  goBack: () => void;
};

export const SingleUser: FC<Props> = ({ user, goBack }) => {
  return (
    <div className="Wrapper">
      <div
        className="CloseBtnWrapper"
        data-testid={'backButton'}
        onClick={goBack}
      >
        <img
          className="CloseBtnIcon"
          alt={'closeBtn'}
          src={
            'https://upload.wikimedia.org/wikipedia/commons/thumb/4/4c/Grey_close_x.svg/512px-Grey_close_x.svg.png?20140617032513'
          }
        />
      </div>
      <div className="SingleUserCardTop">
        <img
          className="SingleUserIcon"
          alt={`user ${user.name}`}
          src={`/foto/${user.photo}`}
        />
        <div data-testid="name">{user.name}</div>
        <div data-testid="position">{user.position}</div>
      </div>
      <div className="SingleUserCardBottom">
        <div className="SingleUserCardBottomTitle">
          <div>Phone</div>
          <div>Nickname</div>
          <div>Email</div>
        </div>
        <div>
          <div>{user.phone}</div>
          <div>{user.nickname}</div>
          <div>{user.email}</div>
        </div>
      </div>
      <div className="BtnSendMessage">Send message</div>
    </div>
  );
};
