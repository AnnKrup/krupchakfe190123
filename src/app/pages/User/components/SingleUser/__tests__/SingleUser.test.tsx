import * as React from 'react';
import { createRenderer } from 'react-test-renderer/shallow';

import { SingleUser } from '../SingleUser';
import { fireEvent, render } from '@testing-library/react';

const renderer = createRenderer();

const props = {
  user: {
    phone: '902-782-3286',
    name: 'Brazil Izair',
    nickname: '@brazilizair',
    email: 'brazil.izair@fallinhay.com',
    position: 'Managing Director',
    photo: '2.jpg',
  },
  goBack: () => {},
};

describe('<UserItem />', () => {
  it('should render and match the snapshot', () => {
    renderer.render(<SingleUser {...props} />);
    const renderedOutput = renderer.getRenderOutput();
    expect(renderedOutput).toMatchSnapshot();
  });

  it('check image', () => {
    const { getByAltText } = render(<SingleUser {...props} />);
    getByAltText(`user ${props.user.name}`);
  });

  it('check user info', () => {
    const { getByTestId } = render(<SingleUser {...props} />);
    const name = getByTestId('name').textContent;
    const position = getByTestId('position').textContent;

    expect(name).toBe(props.user.name);
    expect(position).toBe(props.user.position);
  });

  it('choose goBack fires', () => {
    const goBack = jest.fn();
    const { getByTestId } = render(<SingleUser {...props} goBack={goBack} />);
    const button = getByTestId('backButton');

    fireEvent.click(button);

    expect(goBack).toHaveBeenCalled();
  });
});
