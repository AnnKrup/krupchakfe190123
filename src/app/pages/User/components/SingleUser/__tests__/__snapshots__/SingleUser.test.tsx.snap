// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserItem /> should render and match the snapshot 1`] = `
<div
  className="Wrapper"
>
  <div
    className="CloseBtnWrapper"
    data-testid="backButton"
    onClick={[Function]}
  >
    <img
      alt="closeBtn"
      className="CloseBtnIcon"
      src="https://upload.wikimedia.org/wikipedia/commons/thumb/4/4c/Grey_close_x.svg/512px-Grey_close_x.svg.png?20140617032513"
    />
  </div>
  <div
    className="SingleUserCardTop"
  >
    <img
      alt="user Brazil Izair"
      className="SingleUserIcon"
      src="/foto/2.jpg"
    />
    <div
      data-testid="name"
    >
      Brazil Izair
    </div>
    <div
      data-testid="position"
    >
      Managing Director
    </div>
  </div>
  <div
    className="SingleUserCardBottom"
  >
    <div
      className="SingleUserCardBottomTitle"
    >
      <div>
        Phone
      </div>
      <div>
        Nickname
      </div>
      <div>
        Email
      </div>
    </div>
    <div>
      <div>
        902-782-3286
      </div>
      <div>
        @brazilizair
      </div>
      <div>
        brazil.izair@fallinhay.com
      </div>
    </div>
  </div>
  <div
    className="BtnSendMessage"
  >
    Send message
  </div>
</div>
`;
